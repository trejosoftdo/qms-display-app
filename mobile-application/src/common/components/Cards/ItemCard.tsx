import * as React from 'react';
import { StyleSheet } from 'react-native';
import { Avatar, Card, Text } from 'react-native-paper';
import AppLightTheme from '../../theme';

/**
 * ItemCardProps defines the props for the Cards Component.
 *
 * @property icon - card icon
 * @property title - card title
 * @property onPress - on press card handler
 */
interface ItemCardProps {
  icon: string;
  title: string
  onPress: () => void;
}

/**
 * A component for an item card
 *
 * @param {ItemCardProps} props - The props for the Item Card component.
 */
const ItemCard: React.FC<ItemCardProps> = (props: ItemCardProps) => (
  <Card
    mode="outlined"
    style={styles.container}
    onPress={props.onPress}
  >
    <Card.Content style={styles.content}>
      <Avatar.Icon style={styles.icon} size={64} icon={props.icon} />
      <Text variant="titleMedium" style={styles.title}>{props.title}</Text>
    </Card.Content>
  </Card>
);

const styles = StyleSheet.create({
  container: {
    backgroundColor: AppLightTheme.colors.background,
    marginRight: 2,
    cursor: 'pointer',
    width: '45%',
    maxWidth: 150,
    paddingBottom: 16,
    borderColor: AppLightTheme.colors.outline
  },
  icon: {
    backgroundColor: AppLightTheme.colors.primary,
  },
  content: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  },
  title: {
    color: AppLightTheme.colors.secondary,
    paddingTop: 10,
    textAlign: 'center',
  },
});

export default ItemCard;