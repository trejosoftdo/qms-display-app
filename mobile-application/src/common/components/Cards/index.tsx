import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import ItemCard from './ItemCard';


type Item = {
  id: string;
  name: string;
  icon: string;
  label: string;
};

/**
 * CardsProps defines the props for the Cards Component.
 *
 * @property items - list of items
 * @property onItemSelect - a handler that is called when an item is selected
 */
interface CardsProps {
  items: Item[];
  onItemSelect: (item: Item) => void;
};

/**
 * A component that represents a list of cards
 *
 * @param {CardsProps} props - The props for the Cards component.
 */
const Cards: React.FC<CardsProps> = (props: CardsProps) => (
  <View style={styles.container}>
    {props.items.map(item => (
      <ItemCard
        key={item.name}  
        icon={item.icon}
        title={item.label}
        onPress={() => {
          props.onItemSelect(item);
        }}
      />
    ))}
  </View>
);

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'space-around',
    flexDirection: 'row',
    flexWrap: 'wrap',
    width: '80%',
    paddingBottomBottom: 32,
  },
});

export default Cards;