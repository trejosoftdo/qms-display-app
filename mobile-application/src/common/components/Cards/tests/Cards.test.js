import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Cards from '../';


describe('Cards component', () => {
  const props = {
    onItemSelect: jest.fn(),
    items: [
      {
        id: '1',
        name: 'Test name 1',
        icon: 'lock',
        label: 'Test Label 1',
      },
      {
        id: '2',
        name: 'Test name 2',
        icon: 'information',
        label: 'Test Label 2',
      },
    ]
  };

  it('renders the component with the provided props as expected', () => {
    expect(renderShallow(<Cards {...props} />)).toMatchSnapshot();
  });

  test('calls onPress prop when clicked', () => {
    const { getAllByTestId } = render(
      <Cards {...props} />
    );
    const elements = getAllByTestId('card');
    fireEvent.press(elements[0]);
    expect(props.onItemSelect).toHaveBeenCalled();
  });
});
