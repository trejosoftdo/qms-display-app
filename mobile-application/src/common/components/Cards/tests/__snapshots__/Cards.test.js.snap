// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards component renders the component with the provided props as expected 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "space-around",
      "paddingBottomBottom": 32,
      "width": "80%",
    }
  }
>
  <ItemCard
    icon="lock"
    onPress={[Function]}
    title="Test Label 1"
  />
  <ItemCard
    icon="information"
    onPress={[Function]}
    title="Test Label 2"
  />
</View>
`;
